sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -70,14 +71,22 @@ case class SqlScriptingInterpreter() {
         }
         val dropVariables = variables
           .map(varName => DropVariable(varName, ifExists = true))
-          .map(new SingleStatementExec(_, Origin(), isInternal = true))
+          .map(new SingleStatementExec(_, Origin(), isInternal = true, collectResult = false))
           .reverse
         new CompoundBodyExec(
-          body.collection.map(st => transformTreeIntoExecutable(st)) ++ dropVariables)
+          body.collection.map(st => transformTreeIntoExecutable(st)) ++ dropVariables, session)
       case sparkStatement: SingleStatement =>
         new SingleStatementExec(
           sparkStatement.parsedPlan,
           sparkStatement.origin,
           isInternal = false)
     }
+
+  def execute(executionPlan: Iterator[CompoundStatementExec]): Iterator[Array[Row]] = {
```

Comment:
I would maybe make `buildExecutionPlan` private, and than call it from `execute` - seems like we will never need to call `buildExecutionPlan` from outside?
for example, if we add separate `execute*` function in the future, it can also call `buildExecutionPlan` internally.

Reply:
We are using `buildExecutionPlan` in `SqlScriptingInterpreterSuite`. Should we change the testing logic as well then?

We can call `buildExecutionPlan` from `execute` but can we also leave it to be public?

Reply:
This probably means that we should change the tests a bit, we are simulating `execute` in the tests as well, because before we didn't have it in the interpreter. Now that we have it, we should probably change tests.

Let's do what you said last for now, until other folks review the PR (leave this comment unresolved) and then we can figure if we want to change tests as well.